src/platform-aws.c

```diff
@@ -424,6 +424,25 @@ int platform_init(void)
 			goto exit;
 		}
 	}
+
+	/*
+	 * NCCL v2.19.3 reduced the chunk size used when running NVLS Tree
+	 * algorithm on greater than 4 nodes to 64KiB. This drastically impacted
+	 * performance on AWS (Ref: https://github.com/NVIDIA/nccl/pull/1112/
+	 * for some data). NCCL v2.20.3 has made this a tunable, so revert back
+	 * to using 512KiB which recovers from the regression. Setting this
```

Comment:
This doesn't explain how we deduced that the chunk size was previously 512KiB. In the mentioned PR, I just removed the line that overrides the chunk size in case of >=4 nodes, but I haven't mentioned there that the chunk size was previously 512KiB (It's just the size I have noticed on prints empirically on the benchmark that I have run specifically).

In addition, we should have a task to use this new configuration knob to benchmark the results we get with various chunk sizes on various number of nodes.

Reply:
> (It's just the size I have noticed on prints empirically on the benchmark that I have run specifically

Alright, updating the comment.

> In addition, we should have a task to use this new configuration knob to benchmark the results we get with various chunk sizes on various number of nodes.

Thought you'd never ask. I already did the sweep and this is still the chunksize with the most balanced performance across message sizes (will share some data offline).